drivers/pds/lib/utils.go

```diff
@@ -1313,3 +1314,94 @@ func ValidateAllDataServiceVolumes(deployment *pds.ModelsDeployment, dataService
 	return resourceTemp, storageOp, config, nil
 
 }
+
+func GetPodsFromK8sStatefulSet(deployment *pds.ModelsDeployment, namespace string) ([]corev1.Pod, error) {
+	var ss *v1.StatefulSet
+	err = wait.Poll(maxtimeInterval, timeOut, func() (bool, error) {
+		ss, err = k8sApps.GetStatefulSet(deployment.GetClusterResourceName(), namespace)
+		if err != nil {
+			logrus.Warnf("An Error Occured while getting statefulsets %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		logrus.Errorf("An Error Occured while getting statefulsets %v", err)
+		return nil, err
+	}
+	pods, err := k8sApps.GetStatefulSetPods(ss)
+	if err != nil {
+		logrus.Errorf("An error occured while getting the pods belonging to this statefulset %v", err)
+		return nil, err
+	}
+	return pods, nil
+}
+
+func GetK8sNodeObjectUsingPodName(nodeName string) (*corev1.Node, error) {
+	nodeObject, err := k8sCore.GetNodeByName(nodeName)
+	if err != nil {
+		logrus.Errorf("Could not get the node object for node %v because %v", nodeName, err)
+		return nil, err
+	}
+	return nodeObject, nil
+}
+
+func DrainPxPodOnK8sNode(node *corev1.Node, namespace string) error {
+	labelSelector := map[string]string{"name": "portworx"}
+	pod, err := k8sCore.GetPodsByNodeAndLabels(node.Name, namespace, labelSelector)
+	if err != nil {
+		logrus.Errorf("Could not fetch pods running on the given node %v", err)
+		return err
+	}
+	logrus.Infof("Portworx pod to be drained %v from node %v", pod.Items[0].Name, node.Name)
+	err = k8sCore.DrainPodsFromNode(node.Name, pod.Items, timeOut, maxtimeInterval)
+	if err != nil {
+		logrus.Errorf("Could not drain the node %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func LabelK8sNode(node *corev1.Node, label string) error {
+	keyval := strings.Split(label, "=")
+	err := k8sCore.AddLabelOnNode(node.Name, keyval[0], keyval[1])
+	return err
+}
+
+func RemoveLabelFromK8sNode(node *corev1.Node, label string) error {
+	err := k8sCore.RemoveLabelOnNode(node.Name, label)
+	return err
+}
+
+func UnCordonK8sNode(node *corev1.Node) error {
+	err = wait.Poll(maxtimeInterval, timeOut, func() (bool, error) {
+		err = k8sCore.UnCordonNode(node.Name, timeOut, maxtimeInterval)
+		if err != nil {
+			logrus.Errorf("Failed uncordon node %v due to %v", node.Name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+	return err
+}
```

Comment:
Same here, why not just use already existing `sched-ops` func `k8sCore.UnCordonNode()`? and give it a timeout and interval instead of wrapping it into a whole new func?